### ISSUE TITLE
Show dot or number of unread messages in the list of Apps

### ISSUE THREAD

**Issue by q-wertz** [nextcloud/spreed#5525]
**Is your feature request related to a problem? Please describe.**
As already discussed in a lot of other issues/feature requests one currently misses a lot of notifications/new messages.
E.g. 
- playing sound https://github.com/nextcloud/spreed/issues/4651
- inactive Browser problem https://github.com/nextcloud/spreed/issues/1575 https://github.com/nextcloud/spreed/issues/5321

In my case I also do not see a red dot as prefix in the tab but an asterix * (maybe my browser/OS is the problem?)

**Describe the solution you'd like**
Another additional Notification would maybe be to show a red dot or the number of unread messages as an overlay of the Q in the Apps List

![dot](https://user-images.githubusercontent.com/20948931/116373562-fd684b80-a7fc-11eb-9c46-bf8f6a08dd37.png)
![num](https://user-images.githubusercontent.com/20948931/116373575-ffcaa580-a7fc-11eb-9205-bf2da85021da.png)

**Describe alternatives you've considered**
I am on Talk v10.0.7, in https://github.com/nextcloud/spreed/issues/5321#issuecomment-800190265 it is mentioned that with Talk 11.0.1 some things changed, so maybe notifications are more reliable now?

**Comment by nickvergessen** [comment]
Can't find the ticket atm, but the main issue is that there is no API in the server for this.

The other problem is creating and updating the information. Like counting across all your chats on any page.

**Comment by q-wertz** [comment]
Thanks for the reply

> Can't find the ticket atm, but the main issue is that there is no API in the server for this.

Ah ok. So this is an open issue on the server side? Or not even planned to be implemented there? Might also be useful for some other apps (Deck/Task showing today due cards, ...)

> The other problem is creating and updating the information. Like counting across all your chats on any page.

not a web developer, so I just trust you there on this :smile: 


Is it possible to get something similar into the notifications (bell) sign?
So red dot there and a single message with something like "5 new messages". Can think about some methods (no clue if they would really work technically) depending on what the Notification API is capable of.

**Comment by q-wertz** [comment]
This one does only work inside the app environment? https://github.com/nextcloud/nextcloud-vue/tree/master/src/components/AppNavigationCounter

**Comment by nickvergessen** [comment]
Yeah that is for the left sidebar, not the top one. and as said: populating the "dot/counter" with the information comes with a huge performance penalty.

**Comment by jancborchardt** [comment]
Just for reference: If we do this, we should go for the dot instead of the very detailed counter. (E.g. It doesn’t matter if there’s 12 or 13 unread messages.

**Comment by jancborchardt** [comment]
The server part of this will be implemented with https://github.com/nextcloud/server/pull/26939

**Comment by nickvergessen** [comment]
I'm still worried about the performance impact.
Each app looping over all it's data on each page load will not help.
And pinging in the background even more so.

And when the dot does not appear/vanish the use does not validate the load for me.

**Comment by nickvergessen** [comment]
cc @juliushaertl ^ what's the idea about this

**Comment by juliushaertl** [comment]
We haven't discussed the details about the individual apps yet. One thing I've considered is that we could basically extend the notifications app to just toggle the bubble then depending if there are notifications based on the app id, but not sure yet how feasible that is.

**Comment by q-wertz** [comment]
As https://github.com/nextcloud/server/pull/26939 was merged this should be possible now, right?

**Comment by nickvergessen** [comment]
There is still no way on API level to get the total count in a performant way.

**Comment by RTechSn** [comment]
Is there a ticket for this on their side? Could be that no one knows it is needed.

**Comment by RTechSn** [comment]
> There is still no way on API level to get the total count in a performant way.

I thought this means there is no API for getting the unread number on nextcloud/server side and it blocks future progress on this.
